packages/i18n/src/translateWithMarkup.tsx

```diff
@@ -52,18 +58,22 @@ const VOID_ELEMENTS = [
  * @param {?Object.<string, ReactElement>} mapping Map of tag names to React components.
  * @return {ReactNode} Transformed node.
  */
-export function transformNode(node, mapping = {}) {
-  const { childNodes, localName, nodeType, textContent } = node;
+export function transformNode(
+  node: ChildNode,
+  mapping: Mapping = {}
+): React.ReactNode {
+  const { firstChild, nodeType, textContent } = node;
   if (Node.TEXT_NODE === nodeType) {
     return textContent;
   }
 
-  const children = node.hasChildNodes()
-    ? [...childNodes].map((child) => transform(child, mapping))
-    : null;
```

Comment:
I am pretty sure this old code is wrong, no? It loops over all the child nodes and transforms each, but `transformNode` transform both the given node and all its following siblings, so this old code duplicates a lot of nodes, no?

Well, at least the type system told me that the `children` variable had type `ReactNode[][]`, which definitely seemed wrong.